drivers/note/sched_note.c

```diff
@@ -409,22 +486,31 @@ static void note_spincommon(FAR struct tcb_s *tcb,
                             int type)
 {
   struct note_spinlock_s note;
+  channel_variable_init;
 
   if (!note_isenabled())
     {
       return;
     }
 
+  note_foreach_channel(g_note_channel_list, channel)
```

Comment:
why run the loop twice?

Reply:
In note_foreach_channel, check whether you need to call a custom format function and count the number of times you need to call the nuttx default format function

In channel_foreach_write, write the data formatted by nuttx to the channel with write callback

However, the same channel should not be written twice, fixed